lib/assets/javascripts/utils/filteriseTable.js

```diff
@@ -0,0 +1,46 @@
+/* 
+  filteriseTable adds filter capabilities to an HTML table 
+  table rows are shown/hidden as the user enters text into the filter input field 
+  all rows are made visible when the user clicks the 'clear' icon
+*/
+(function(ctx){
+
+  var filter = (function(el){
+    var query = $(el).val(),
+        regex = new RegExp(query, 'i');
+
+    $.each($(el).closest("table").find("tbody tr"), function(idx, tr){
+      if(regex.test($(tr).text())){
+        $(tr).show();
+      }else{
+        $(tr).hide();
+      }
+    });
+  });
+
+  var clear = (function(el){
+    $(el).val('');
+    $(el).closest("table").find("tbody tr").show();
+  });
+
+  ctx.init = ctx.init || (function(options){
+    if($ && options && options.selector){
+      var id = $(this).attr("id");
+
+      /* initialize a debounced listener for the filter box */
+      var debounced = dmproadmap.utils.debounce(filter);
+
+      /* Bind the clear function to the clear icon's click event */
+      $(options.selector).keyup(function(){
+        debounced(this, 250); 
```

Comment:
the call to debounced function internally calls to the function that you have passed in L31, i.e. the filter function and passes the arguments this (e.g. the current HTML element) together with the integer 250. The second argument is never used in filter function, which only has el as argument, so for is misleading here. I guess you wanted to decrease the time in which the filter function would be triggered. I resolved some conflicts from this PR together with cleaning the call to debounced.